### PR TITLE
[km200] Fix item type for dynamic channel type

### PR DESCRIPTION
**Pull request by lolodomo**
Related to #12712

Signed-off-by: Laurent Garnier <lg.hc@free.fr>

### CODE REVIEW

File: bundles/org.openhab.binding.km200/src/main/java/org/openhab/binding/km200/internal/handler/KM200ThingHandler.java [hunk 1]
```diff
@@ -156,10 +156,10 @@ Channel createChannel(ChannelTypeUID channelTypeUID, ChannelUID channelUID, Stri
         String itemType = "";
         String category = null;
         if (CoreItemFactory.NUMBER.equals(type)) {
-            itemType = "NumberType";
+            itemType = CoreItemFactory.NUMBER;
```
Comment (jlaur) [inline bundles/org.openhab.binding.km200/src/main/java/org/openhab/binding/km200/internal/handler/KM200ThingHandler.java:159]: Now local variable `itemType` could be dropped and `type` can be used directly in line 170 and 178.

Comment (lolodomo) [reply bundles/org.openhab.binding.km200/src/main/java/org/openhab/binding/km200/internal/handler/KM200ThingHandler.java:159]: Yes, certainly

Comment (lolodomo) [reply bundles/org.openhab.binding.km200/src/main/java/org/openhab/binding/km200/internal/handler/KM200ThingHandler.java]: Done

**Review by jlaur — APPROVED** [review]
LGTM

**Pull request merged**